examples/raspberrypi/rp2xxx/src/rp2040_only/hd44780.zig

```diff
@@ -0,0 +1,76 @@
+const std = @import("std");
+const microzig = @import("microzig");
+const drivers = microzig.drivers;
+const lcd = drivers.display.HD44780;
+const lcd_config = drivers.display.hd44780.Config;
+const io_device = drivers.base.Datagram_Device;
+const WriteError = io_device.WriteError;
+
+const rp2040 = microzig.hal;
+const i2c = rp2040.i2c;
+const gpio = rp2040.gpio;
+const peripherals = microzig.chip.peripherals;
+const timer = rp2040.time;
+
+const i2c0 = i2c.instance.num(0);
+
+pub const i2c_device = struct {
+    i2c_bus: i2c.I2C,
+    addr: i2c.Address,
+    pub fn init(bus: i2c.I2C, addr: i2c.Address) i2c_device {
+        return .{
+            .i2c_bus = bus,
+            .addr = addr,
+        };
+    }
+
+    fn writev(ctx: *anyopaque, datagrams: []const []const u8) WriteError!void {
+        const device: *i2c_device = @ptrCast(@alignCast(ctx));
+        device.i2c_bus.writev_blocking(device.addr, datagrams, null) catch return WriteError.IoError;
+    }
+
+    const vtable = drivers.base.Datagram_Device.VTable{
+        .connect_fn = null,
+        .disconnect_fn = null,
+        .writev_fn = i2c_device.writev,
+        .readv_fn = null,
+    };
+
+    pub fn datagram_device(device: *i2c_device) io_device {
+        return io_device{
+            .object = device,
+            .vtable = &vtable,
+        };
+    }
+};
```

Comment:
We do have a ready-to-use implementation in https://github.com/ZigEmbeddedGroup/microzig/blob/dad43d5f9bad5bf8082dd2425c469e323f2066d7/port/raspberrypi/rp2xxx/src/hal/drivers.zig#L18-L34